[memgraph-2.12 < T] Improve global and query memory limit (#284)
memgraph/documentation: pages/fundamentals/storage-memory-usage.mdx

### Control query memory usage

Query execution also uses up RAM. All allocations inside query are counted

> from intermediary results to allocations made while creating vertices and edges.

Reviewer: ```suggestion
from intermediary results to allocations made while creating nodes and relationships.
```

```

This query tries to clean up as much unused memory as possible without affecting
currently running transactions.

## Virtual memory

> Memgraph uses a jemalloc allocator to handle all allocations and to track allocations happening 
> in system.
> 
> The default operating system limits on mmap counts is likely to be too low, which may result in out of memory exceptions.

Reviewer: ```suggestion
Memgraph uses a [`jemalloc`](https://jemalloc.net/) allocator to handle all allocations and to track allocations happening in the system.

The default operating system limits on `mmap` counts are likely to be too low, which may result in out-of-memory exceptions.
```

> 

Reviewer: ```suggestion
To find out the current `mmap` count of a running Memgraph process use the following command, where `$PID` is the process ID of the running Memgraph process:

```bash
wc -l /proc/$PID/maps
```
```

> Memgraph uses a jemalloc allocator to handle all allocations and to track allocations happening 
> in system.
> 
> The default operating system limits on mmap counts is likely to be too low, which may result in out of memory exceptions.
> 
> On Linux, you can increase the limits by running the following command as root:
> 
> ```bash
> sysctl -w vm.max_map_count=262144
> ```
> 
> To set this value permanently, update the vm.max_map_count setting in /etc/sysctl.conf. To verify after rebooting, run sysctl vm.max_map_count.
> 
> The RPM and Debian packages will configure this setting automatically. No further configuration is required.
> 
> You can find out the current mmap count of a running Memgraph process using the following command, where $PID is the process ID of the running Memgraph process:
> 
> ```bash
> wc -l /proc/$PID/maps
> ```

Reviewer: ```suggestion
Memgraph uses a [`jemalloc`](https://jemalloc.net/) allocator to handle all allocations and to track allocations happening in the system.

The default operating system limits on `mmap` counts are likely to be too low, which may result in out-of-memory exceptions.

To find out the current `mmap` count of a running Memgraph process use the following command, where `$PID` is the process ID of the running Memgraph process:

```bash
wc -l /proc/$PID/maps
```

On Linux, you can increase the limits by running the following command as `root`:

```bash
sysctl -w vm.max_map_count=262144
```

To increase this value permanently, update the `vm.max_map_count` setting in */etc/sysctl.conf*. To verify, reboot the system and run `sysctl vm.max_map_count`.

The RPM and Debian packages will configure this setting automatically so you don't need to do any more configurations.
```
